darkcluster/src/main/java/com/linkedin/darkcluster/impl/DarkClusterManagerImpl.java

```diff
@@ -127,10 +127,13 @@ public boolean handleDarkRequest(RestRequest originalRequest, RequestContext ori
         DarkClusterConfigMap configMap = _facilities.getClusterInfoProvider().getDarkClusterConfigMap(_sourceClusterName);
         for (String darkClusterName : configMap.keySet())
         {
-          RestRequest newD2Request = rewriteRequest(reqCopy, darkClusterName);
-          // now find the strategy appropriate for each dark cluster
-          DarkClusterStrategy strategy = _darkClusterStrategyFactory.get(darkClusterName);
-          darkRequestSent = strategy.handleRequest(reqCopy, newD2Request, newRequestContext);
+          if (_darkGateKeeper.shouldDispatchToDark(originalRequest, originalRequestContext, darkClusterName))
```

Comment:
Do we still need the (_darkGateKeeper.shouldDispatchToDark(originalRequest, originalRequestContext) outside of the for loop. Since we need to check shouldDispatchToDark with the dark cluster name anyway.
Also should we add unit tests for this new logic?

Reply:
I leave it there for maintaining backward compatibility. If the service owner is on a older version of pegasus, they don't need to change their DarkGateKeeperImpl to migrate to this new interface.

I'll add a unit test later. Thanks for reviewing

Reply:
You're right, we don't need the check outside the for loop. I can actually deprecate the old API, as the use case of it can be fully covered by the new API.

Also added unit tests